bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -616,6 +617,23 @@ public void safeRun() {
 
     void checkForFaultyBookies() {
         List<BookieId> faultyBookies = bookieClient.getFaultyBookies();
+        if (faultyBookies.isEmpty()) {
+            return;
+        }
+
+        boolean isEnable = false;
+        try {
+            isEnable = metadataDriver.isHealthCheckEnabled().get();
+        } catch (InterruptedException e) {
+            LOG.error("isHealthCheckEnabled throw InterruptedException", e);
+        } catch (ExecutionException e) {
+            LOG.error("isHealthCheckEnabled throw ExecutionException", e);
```

Comment:
`LOG.error("Cannot verify if healthcheck is enabled", e.getCause());`

we can log "e.getCause()" because ExecutionException is always a wrapper for another problem

Reply:
OK I will fix